docs/internals/events.md

```diff
@@ -0,0 +1,135 @@
+# Events
+
+As described in the [architecture](./architecture.md) events play a key role in the complete system: they relay all information between seperate components (like plugins). This makes the codebase scalable, testable and it seperates concerns.
+
+if you run the Gekko UI events are relayed between core components as well as broadcasted (via the UI server) to the web UI. This means that all events broadcasted by any plugin automatically end up in the web UI.
+
+Note that all events from Gekko come from a plugin (with the exception of the `candle` event, which comes from the market), and no plugin is required for Gekko to run, this means it might be possible that some events are never broadcasted since their originating plugin is not active. If a plugin wants to listen to an event that will never be broadcasted (because of a lack of another plugin) this will be warned in the console like so:
+
+    (WARN): Paper Trader wanted to listen to the tradingAdvisor, however the tradingAdvisor is disabled.
```

Comment:
Are the extra spaces here intended?

Reply:
Yes! If you add 4 whitepaces in markdown it will convert it into a codeblock, see here for an example: 

https://github.com/askmike/gekko/blame/develop/docs/strategies/creating_a_strategy.md#L13

&

https://gekko.wizb.it/docs/strategies/creating_a_strategy.html#Boilerplate-strategies

Reply:
Lol. Somehow I didn't think of the possibility that you wanted it to be a codeblock on purpose...